From 28b91a203b21b4ffd8d438bd62888419c95d6dbd Mon Sep 17 00:00:00 2001
From: Aleksandr Gumroian <aleksandr.gumroian@nic.cz>
Date: Tue, 16 Nov 2021 17:35:39 +0100
Subject: [PATCH] Fix time expiration for static leases in both DHCP tables

(cherry picked from commit a34df8bc8ac4562eb50277e89828cdbd87fb8019)
---
 js/src/common/network/DHCP6ClientsList.js | 4 +++-
 js/src/common/network/DHCPClientsList.js  | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/js/src/common/network/DHCP6ClientsList.js b/js/src/common/network/DHCP6ClientsList.js
index 4147b419..e2a262b0 100644
--- a/js/src/common/network/DHCP6ClientsList.js
+++ b/js/src/common/network/DHCP6ClientsList.js
@@ -71,7 +71,9 @@ function DHCP6ClientsListItem({ ipv6, expires, duid, hostname }) {
             <td>{ipv6}</td>
             <td>{duid}</td>
             <td className="text-center">
-                {moment.unix(expires).format("YYYY-MM-DD HH:mm")}
+                {expires
+                    ? moment.unix(expires).format("YYYY-MM-DD HH:mm")
+                    : _("Never")}
             </td>
         </tr>
     );
diff --git a/js/src/common/network/DHCPClientsList.js b/js/src/common/network/DHCPClientsList.js
index 1013384d..a10a68a3 100644
--- a/js/src/common/network/DHCPClientsList.js
+++ b/js/src/common/network/DHCPClientsList.js
@@ -69,7 +69,9 @@ function DHCPClientsListItem({ ip, expires, mac, hostname, active }) {
             <td>{ip}</td>
             <td>{mac}</td>
             <td className="text-center">
-                {moment.unix(expires).format("YYYY-MM-DD HH:mm")}
+                {expires
+                    ? moment.unix(expires).format("YYYY-MM-DD HH:mm")
+                    : _("Never")}
             </td>
             <td className="text-center">
                 <i
-- 
2.32.0

